support/src/main/java/org/jenkinsci/plugins/workflow/support/storage/SimpleXStreamFlowNodeStorage.java

```diff
@@ -160,12 +164,16 @@ public boolean canConvert(Class type) {
      * To group node and their actions together into one object.
      */
     private static class Tag {
-        FlowNode node;
-        Action[] actions;
+        final @Nonnull FlowNode node;
+        private final @CheckForNull Action[] actions;
 
-        private Tag(FlowNode node, List<Action> actions) {
+        private Tag(@Nonnull FlowNode node, @Nonnull List<Action> actions) {
             this.node = node;
-            this.actions = actions.toArray(new Action[actions.size()]);
+            this.actions = actions.isEmpty() ? null : actions.toArray(new Action[actions.size()]);
+        }
+
+        public @Nonnull List<Action> actions() {
+            return actions != null ? Arrays.asList(actions) : Collections.<Action>emptyList();
```

Comment:
To be clear, the _original_ code was already converting `List<Action>` to `Action[]`. The only real change I am making here is replacing `new Action[0]` with `null`. The other changes merely encapsulate the null check when serving the list back.